cmd/preflight/cmd/root_test.go

```diff
@@ -85,17 +87,45 @@ var _ = Describe("cmd package utility functions", func() {
 				BeforeEach(func() {
 					os.Setenv("PFLT_LOGFILE", "/tmp/foo.log")
 					os.Setenv("PFLT_LOGLEVEL", "trace")
+					DeferCleanup(func() {
+						os.Unsetenv("PFLT_LOGFILE")
+						os.Unsetenv("PFLT_LOGLEVEL")
+					})
 				})
 				It("should have overrides in place", func() {
-					initConfig()
+					initConfig(viper.Instance())
 					Expect(viper.Instance().GetString("namespace")).To(Equal(DefaultNamespace))
 					Expect(viper.Instance().GetString("artifacts")).To(Equal(artifacts.DefaultArtifactsDir))
 					Expect(viper.Instance().GetString("logfile")).To(Equal("/tmp/foo.log"))
 					Expect(viper.Instance().GetString("loglevel")).To(Equal("trace"))
 				})
-				AfterEach(func() {
-					os.Unsetenv("PFLT_LOGFILE")
-					os.Unsetenv("PFLT_LOGLEVEL")
+			})
+			When("a config file is present", func() {
+				var viper *spfviper.Viper
+				BeforeEach(func() {
+					viper = spfviper.New()
+					fs := afero.NewMemMapFs()
+					viper.SetFs(fs)
+					configFile := `
+namespace: configspace
+artifacts: configartifacts
+logfile: configlogfile
+loglevel: configloglevel
+`
+					// This has to be written to current working directory in the memmapfs, since '.'
+					// as the config path is translated into an absolute path prior to the config
+					// file being read. So, here, we just create the file at that path, it just
+					// happens to be in a memmapfs
+					cwd, err := os.Getwd()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(afero.WriteFile(fs, filepath.Join(cwd, "config.yaml"), bytes.NewBufferString(configFile).Bytes(), 0o644)).To(Succeed())
+				})
+				It("should read all of the config", func() {
+					initConfig(viper)
```

Comment:
while we're here, can we rename this? I would (and do currently) have no clue what this as I read it here in this context.

Reply:
Assume you meant the It() clause? Done.